src/test/java/com/puppycrawl/tools/checkstyle/checks/OrderedPropertiesCheckTest.java

```diff
@@ -0,0 +1,228 @@
+    public void testKeysOnly() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(OrderedPropertiesCheck.class);
+        final String[] expected = {
+            "3: " + getCheckMessage(MSG_KEY, "key1", "key2"),
+        };
+        verify(checkConfig, getPath("InputOrderedProperties1OrderKey.properties"), expected);
+    }
+
+    @Test
+    public void testEmptyKeys() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(OrderedPropertiesCheck.class);
+        final String[] expected = {
+            "3: " + getCheckMessage(MSG_KEY, "key11", "key2"),
+        };
+        verify(checkConfig, getPath("InputOrderedProperties2EmptyValue.properties"), expected);
+    }
+
+    @Test
+    public void testBinaryOrder() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(OrderedPropertiesCheck.class);
+        final String[] expected = {
+            "5: " + getCheckMessage(MSG_KEY, "key", "Ã¤"),
+        };
+        verify(checkConfig, getPath("InputOrderedProperties3BinaryOrder.properties"), expected);
+    }
+
+    @Test
+    public void testCommentsMultiLine() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(OrderedPropertiesCheck.class);
+        final String[] expected = {
+            "5: " + getCheckMessage(MSG_KEY, "aKey", "multi.line"),
+        };
+        verify(checkConfig, getPath("InputOrderedProperties5CommentsMultiLine.properties"),
+                expected);
+    }
+
+    @Test
+    public void testShouldNotProcessFilesWithWrongFileExtension() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(OrderedPropertiesCheck.class);
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+        verify(checkConfig, getPath("InputOrderedProperties.txt"), expected);
+    }
+
+    @Test
+    public void testIoException() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(OrderedPropertiesCheck.class);
+        final OrderedPropertiesCheck check = new OrderedPropertiesCheck();
+        check.configure(checkConfig);
+        final String fileName =
+                getPath("InputOrderedPropertiesCheckNotExisting.properties");
+        final File file = new File(fileName);
+        final FileText fileText = new FileText(file, Collections.emptyList());
+        try {
+            final SortedSet<LocalizedMessage> messages =
+                    check.process(file, fileText);
+            fail("Expected an CheckstyleException to be thrown : " + messages);
+        }
+        catch (CheckstyleException anCheckstyleException) {
+            final String causeMessageStart = "Unable to open file";
+            assertTrue("Unable to open file ",
+                    anCheckstyleException.getMessage().startsWith(causeMessageStart));
+        }
+    }
+
+    @Test
+    public void testWrongKeyTypeInProperties() throws Exception {
```

Comment:
Question wasn't responded at https://github.com/checkstyle/checkstyle/pull/6343#discussion_r245652527
> Why can't this use an input file instead of hacking the check?

Reply:
I think I did some complicated stuff here to get the test coverage green. I think it is not required any more, because I removed the if from the put method.
I will rework this.

Reply:
Test deleted. Done.